### ISSUE TITLE
Scan for Kit: removal of custom kits

### ISSUE THREAD

**Issue by hokiedsp** [vector-of-bool/vscode-cmake-tools#401]
### Brief Issue Summary

[The documentation](https://vector-of-bool.github.io/docs/vscode-cmake-tools/kits.html) states that

```
To prevent custom kits from being overwritten, give them unique names. CMake Tools will not 
delete entries from cmake-kits.json, only add and update existing ones.
```

This feature seems to be supported partially. I'm trying to set up toolchains for TI microcontroller build and noticing some of its compiler settings removed after rescan. (Now should I set it as compiler or toolchain kit is whole another question that I'll need to look into later but I think my observation stands regardless of kit type.)

### Expected:

My customized `cmake-kits.json` file before rescan:

```
[
  {
    "name": "MSP430 GCC 6.2.1.16",
    "compilers": {
      "CC": "C:/ti/ccsv7/tools/compiler/ti-cgt-msp430_16.9.6.LTS/bin/cl430.exe",
      "CXX": "C:/ti/ccsv7/tools/compiler/ti-cgt-msp430_16.9.6.LTS/bin/cl430.exe"
    }
  },
  {
    "name": "MSP430 GCC 6.4.0.32",
    "compilers": {
      "CC": "C:/ti/ccsv7/tools/compiler/msp430-gcc-6.4.0.32_win32/bin/msp430-elf-gcc.exe",
      "CXX": "C:/ti/ccsv7/tools/compiler/msp430-gcc-6.4.0.32_win32/bin/msp430-elf-g++.exe"
    }
  },
  {
    "name": "MSP430 TI 4.4.8",
    "compilers": {
      "CC": "C:/ti/ccsv7/tools/compiler/ti-cgt-msp430_4.4.8/bin/cl430.exe",
      "CXX": "C:/ti/ccsv7/tools/compiler/ti-cgt-msp430_4.4.8/bin/cl430.exe"
    }
  },
  {
    "name": "MSP430 TI 15.12.4",
    "compilers": {
      "CC": "C:/ti/ccsv7/tools/compiler/ti-cgt-msp430_15.12.4.LTS/bin/cl430.exe",
      "CXX": "C:/ti/ccsv7/tools/compiler/ti-cgt-msp430_15.12.4.LTS/bin/cl430.exe"
    }
  },
  {
    "name": "MSP430 TI 16.9.7",
    "compilers": {
      "CC": "C:/ti/ccsv7/tools/compiler/ti-cgt-msp430_16.9.7.LTS/bin/cl430.exe",
      "CXX": "C:/ti/ccsv7/tools/compiler/ti-cgt-msp430_16.9.7.LTS/bin/cl430.exe"
    }
  },
  {
    "name": "Visual Studio Build Tools 2017 - amd64",
    "visualStudio": "4fdde432",
    "visualStudioArchitecture": "amd64",
    "preferredGenerator": {
      "name": "Visual Studio 15 2017",
      "platform": "x64"
    }
  },
  {
    "name": "Visual Studio Build Tools 2017 - x86",
    "visualStudio": "4fdde432",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 15 2017"
    }
  }
]
```

All entries have different names and compiler paths, so I expect rescanning for kits to make no changes to the list.

### Apparent Behavior:

After rescan, I get

```
[
  {
    "name": "MSP430 GCC 6.2.1.16",
    "compilers": {
      "CC": "C:/ti/ccsv7/tools/compiler/ti-cgt-msp430_16.9.6.LTS/bin/cl430.exe",
      "CXX": "C:/ti/ccsv7/tools/compiler/ti-cgt-msp430_16.9.6.LTS/bin/cl430.exe"
    }
  },
  {
    "name": "MSP430 GCC 6.4.0.32",
    "compilers": {
      "CC": "C:/ti/ccsv7/tools/compiler/msp430-gcc-6.4.0.32_win32/bin/msp430-elf-gcc.exe",
      "CXX": "C:/ti/ccsv7/tools/compiler/msp430-gcc-6.4.0.32_win32/bin/msp430-elf-g++.exe"
    }
  },
  {
    "name": "MSP430 TI 16.9.7",
    "compilers": {
      "CC": "C:/ti/ccsv7/tools/compiler/ti-cgt-msp430_16.9.7.LTS/bin/cl430.exe",
      "CXX": "C:/ti/ccsv7/tools/compiler/ti-cgt-msp430_16.9.7.LTS/bin/cl430.exe"
    }
  },
  {
    "name": "Visual Studio Build Tools 2017 - amd64",
    "visualStudio": "4fdde432",
    "visualStudioArchitecture": "amd64",
    "preferredGenerator": {
      "name": "Visual Studio 15 2017",
      "platform": "x64"
    }
  },
  {
    "name": "Visual Studio Build Tools 2017 - x86",
    "visualStudio": "4fdde432",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 15 2017"
    }
  }
]
```

The older TI compilers (4.8.4 and 15.12.4) disappear.

### CMake/Build Log

```
[kit] Found Kit: Visual Studio Build Tools 2017 - x86
[kit] Found Kit: Visual Studio Build Tools 2017 - amd64
[kit] Successfully loaded 5 kits from C:\Users\tikuma\AppData\Roaming\CMakeTools\cmake-kits.json
```

Nothing is logged on `log.txt` for this action.

### Platform and Versions

[[For issues, fill out the following. Otherwise delete this section.]]

- **Operating System**: Windows 7
- **CMake Version**: 3.11.1
- **VSCode Version**: 1.22.2
- **CMake Tools Extension Version**: 0.11-beta.5
- **Compiler/Toolchain**: Visual C++ 2017 + TI CCS supplied compilers

**Comment by vector-of-bool** [comment]
Thanks for the details. It'll make reproducing the case easier. I'll look at this and ping back if I need any additional details.

**Comment by vector-of-bool** [comment]
@hokiedsp does this seem the same as #451? It's possible that the file-watcher isn't loading your kits in time for CMake Tools to know about them.

**Comment by vector-of-bool** [comment]
I'll mark this closed from #452. It'll be fixed in 1.0.1, but re-open if it continues to occur after that.
